### ISSUE TITLE
Gallery access popup never shows on android and code returns declined

### ISSUE THREAD

**Issue by aolesiakmini** [yasirkula/UnityNativeGallery#235]
**Description of the bug**

When I try to get photo from gallery (or take a photo using camera plugin), the game becomes unresponsive for a few seconds and after that it returns result == declined. I never see prompt for permission.

The problem is only on android. On iOS everything works fine.

I've added some debug logs in there. I see logs 1, 2, 3, 4, 5(this one only first time), Result.Denied each time I try to get photo from gallery.
![image](https://user-images.githubusercontent.com/79532091/143468218-ae23d04e-54ef-4fd8-9850-c21f28a5e524.png)

I've tried setting Write Permission to External (SDCard) in Player Settings, it did not change anything.

**Platform specs**

- Unity version: 2019.4.30f1
- Platform: Android
- Device: motorola one action and a few other android phones
- How did you download the plugin: Asset store

My app has minimum API level set to 24(Anroid 7.0 Nougat) and target API level set to 30(Android 11.0), not sure if that is important

Logcat:
![image](https://user-images.githubusercontent.com/79532091/143482279-c56426d2-c52b-4eaa-8814-2daf89bd284d.png)

Do you have any ideas what could be wrong?

**Comment by yasirkula** [comment]
Does changing `PlayerPrefs.GetInt( "NativeGalleryPermission", (int) Permission.ShouldAsk )` to `(int) Permission.ShouldAsk` has any effect by any chance?

**Comment by aolesiakmini** [comment]
Tried it, it didn't help, but in the meantime I think I figured out my problem and it had nothing to do with the actual plugin code.

My android manifest contained those lines:
<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:maxSdkVersion="18" tools:node="remove" />
  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" android:maxSdkVersion="18" tools:node="remove" />
  
If i understand android manifests correctly, those lines make it impossible for app to request read/write permissions.
  
They were added a long time ago, because some 3rd party code was requesting read/write permission at app start, which was very bad, because some players never gave the game a chance after seeing this popup.
  
I'll be trying to find what 3rd party code is asking for permission and then try to cut that piece out.

Closing the issue, thank you for trying to help
